client/rpc/wallet.ts

```diff
@@ -0,0 +1,57 @@
+import { createWalletClient, custom, WalletClient } from "viem";
+import { sepolia } from "viem/chains";
+import { getFromStore, StorageKey } from "./storage";
+let walletClient: any;
+
+const addChainIfMissing = async (walletClient: WalletClient) => {
+  const mruInfo = getFromStore(StorageKey.MRU_INFO);
+  const { domain, rpcUrls } = mruInfo;
+
+  const currentChainId = await walletClient.getChainId();
+  const domainChainId = domain.chainId;
+  const chain = {
+    ...sepolia,
+    name: "Stackr Devnet",
+    rpcUrls: {
+      default: {
+        http: rpcUrls,
+      },
+    },
+    id: domainChainId,
+  };
+  if (currentChainId !== chain.id) {
+    try {
+      await walletClient.switchChain({ id: chain.id });
+    } catch (e) {
+      console.log(e);
+      await walletClient.addChain({
+        chain,
+      });
+    }
+  }
+};
+
+export const getWalletClient = async (): Promise<WalletClient> => {
+  const eth = (window as any).ethereum;
+  if (eth == null) {
+    console.log("Wallet injector not installed; using read-only defaults");
```

Comment:
Getting this error with Rabby/Metamask on Chrome

Reply:
I think you need to request a wallet connection before getting wallet client

Reply:
if you are getting this error you are not running it on localhost

Reply:
index.html file won't work with wallets, host it on localhost